projects/driver_controls/src/x86/ads1015.c

```diff
@@ -1,28 +1,127 @@
+// This module emulates the behavior of ADS1015 on x86.
+// Instead of having interrupts raised by the device once a channel finishes conversion,
+// there is a soft timer with prv_timer_callback that is called periodically to update channel
+// readings and switch to the next channel. The rotation of the channels is implemented using
+// bitsets. The readings are just a arbitrary value.
 #include "ads1015.h"
-#include <status.h>
-#include <stdio.h>
+#include <string.h>
 #include "ads1015_def.h"
-#include "gpio_it.h"
+#include "soft_timer.h"
 
-// Initiates ads1015 by setting up registers and enabling ALRT/RDY Pin.
+#define ADS1015_CHANNEL_UPDATE_PERIOD 10
+#define ADS1015_CHANNEL_ARBITRARY_READING 0
+
+// Checks if a channel is enabled (true) or disabled (false).
+static bool prv_channel_is_enabled(Ads1015Storage *storage, Ads1015Channel channel) {
+  if (storage == NULL || channel >= NUM_ADS1015_CHANNELS) {
```

Comment:
Nit: `#include <stddef.h>` for `NULL`. Also be consistent about the order of checks in a guard. This and the function below are inconsistent.

Reply:
NULL must be already included in one of the libraries since the code compiles?

Reply:
Yes, but it's good practice to explicitly include it. Otherwise, if we didn't include the header which happened to include `stddef.h`, your code would now fail to build.